### ISSUE TITLE
Please don't use the version field to differentiate parallel artifacts

### ISSUE THREAD

**Issue by nafg** [zio/interop-cats#592]
The difference between interop between CE2-ZIO2 and CE3-ZIO2 etc. is not that one is a newer version than the other. They are separate artifacts. They should have separate artifact names.

The version field should only be used when the _same_ artifact _evolves_.

Otherwise tools like Scala Steward, sbt-updates, Renovate, etc. become useless. And they are very valuable.

**Comment by tschuchortdev** [comment]
Yes, please! I just ran into this problem. [Section 10 of the SemVer spec](https://semver.org/#spec-item-10) mandates that metadata like this should be appended with a plus sign, which automated tools will ignore.

**Comment by nafg** [comment]
It's not metadata. They are different artifacts. They are parallel paths of development.

**Comment by nafg** [comment]
This is the same way that cross publishing for Scala versions and Scala.js versions works.

**Comment by neko-kai** [comment]
@nafg @tschuchortdev 

I almost agree that it would be more convenient to have them as separate artifacts instead of versions, but then the different series aren't honestly 'separate' artifacts. You can't use more than one of them at the same time because you can't use multiple versions of ZIO or cats-effect at the same time.

Both ZIO and cats-effect have made the same choice to retain the same artifact names while doing breaking rewrites AND they keep releasing new versions in their different respective version series – that too breaks Scala Steward and others – Steward keeps bothering you to upgrade to ZIO2 or CE3 if you keep using ZIO1 or CE2, the way to fix that nagging is to use steward's settings to block updates to new major versions – it supports that. The nagging is somewhat intentional too, you have to eventually upgrade to the latest versions of these libraries - it's unlikely the old versions will be well supported forever.

**Comment by nafg** [comment]
>  The nagging is somewhat intentional too, you have to eventually upgrade to the latest versions of these libraries

That is **precisely** the point. In the case of ZIO itself (and many more examples), there is a single ordering 1.0.17 < 1.0.18 < 2.0.0 < 2.0.1 etc. If I have code that uses ZIO 1.0 and I can't be bothered to migrate it yet, that's fine but the **whole point** of Scala Steward is to nag me about that.

But if I'm using zio-interop-cats 12.0.0.1 then when it comes to zio-interop-cats I *only* want upgrades to 12.0.0.2. Version 23.0.0.0 is not an evolution of 12.0.0.2, it's a different use case. And they aren't version numbers because they don't follow an ordering. 23.0.0.0 is not "after" 12.0.0.2, for instance.

But instead, Scala Steward *will not tell me about 12.0.0.2*. It will only tell me about 23.0.0.2. There is no reason it should be like that.

It is no different than publishing for different scala versions. If I'm on Scala 2 I don't want Scala Steward to give me the Scala 3 version of ZIO. Improvements to ZIO-for-scala-2 come out all the time and Scala Steward knows not to give me the Scala 3 artifact because they are implemented as parallel artifacts.

I can see the benefit of using git functionality to keep the parallel lines of development in sync, but that just underscores that they are parallel. I suppose it would be more complicated to have separate artifact names with a common version but I could help set that up. Also, it doesn't have to be a common version -- you could continue using the same exact version scheme but just use a slightly different artifact name on each branch.
